packages/angular/cli/commands/update-impl.ts

```diff
@@ -5,62 +5,268 @@
-
-  async parseArguments(schematicOptions: string[], schema: Option[]): Promise<Arguments> {
-    const args = await super.parseArguments(schematicOptions, schema);
-    const maybeArgsLeftovers = args['--'];
-
-    if (maybeArgsLeftovers
-        && maybeArgsLeftovers.length == 1
-        && maybeArgsLeftovers[0] == '@angular/cli'
-        && args.migrateOnly === undefined
-        && args.from === undefined) {
-      // Check for a 1.7 angular-cli.json file.
-      const oldConfigFileNames = [
-        normalize('.angular-cli.json'),
-        normalize('angular-cli.json'),
-      ];
-      const oldConfigFilePath = findUp(oldConfigFileNames, process.cwd())
-                             || findUp(oldConfigFileNames, __dirname);
-
-      if (oldConfigFilePath) {
-        args.migrateOnly = true;
-        args.from = '1.0.0';
+  async parseArguments(_schematicOptions: string[], _schema: Option[]): Promise<Arguments> {
+    return {};
+  }
+
+  // tslint:disable-next-line:no-big-function
+  async run(options: UpdateCommandSchema & Arguments) {
+    const packages: PackageIdentifier[] = [];
+    for (const request of options['--'] || []) {
+      try {
+        const packageIdentifier: PackageIdentifier = npa(request);
+
+        // only registry identifiers are supported
+        if (!packageIdentifier.registry) {
+          this.logger.error(`Package '${request}' is not a registry package identifer.`);
+
+          return 1;
+        }
+
+        if (packages.some(v => v.name === packageIdentifier.name)) {
+          this.logger.error(`Duplicate package '${packageIdentifier.name}' specified.`);
+
+          return 1;
+        }
+
+        // If next option is used and no specifier supplied, use next tag
+        if (options.next && !packageIdentifier.rawSpec) {
+          packageIdentifier.fetchSpec = 'next';
+        }
+
+        packages.push(packageIdentifier);
+      } catch (e) {
+        this.logger.error(e.message);
+
+        return 1;
       }
     }
 
-    // Move `--` to packages.
-    if (args.packages == undefined && args['--']) {
-      args.packages = args['--'];
-      delete args['--'];
-    }
+    if (options.all && packages.length > 0) {
+      this.logger.error('Cannot specify packages when using the "all" option.');
 
-    return args;
-  }
+      return 1;
+    } else if (options.all && options.migrateOnly) {
+      this.logger.error('Cannot use "all" option with "migrate-only" option.');
+
+      return 1;
+    } else if (!options.migrateOnly && (options.from || options.to)) {
+      this.logger.error('Can only use "from" or "to" options with "migrate-only" option.');
+
+      return 1;
+    }
 
-  async run(options: UpdateCommandSchema & Arguments) {
     const packageManager = getPackageManager(this.workspace.root);
+    this.logger.info(`Using package manager: '${packageManager}'`);
+
+    if (options.all || packages.length === 0) {
+      // Either update all packages or show status
+      return this.runSchematic({
```

Comment:
Why do we have two `force` options?

Reply:
The first is the schematic runtime force option which allows overwriting of already existing files on create actions (https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/schematics/src/sink/host.ts#L31)

The second is the update schematic's option.